### ISSUE TITLE
Missing typecheck error when using aggregator functions in emit statements

### ISSUE THREAD

**Issue by psybers** [boalang/compiler#258]
The following code:

````
o: output sum of int;
sum << 1;
````

Should generate a typecheck error on the 2nd line.  Instead no errors are created, leading to a runtime issue because it generates this line `context.write(new boa.io.EmitKey("sum"), new boa.io.EmitValue(1l));`
